java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -162,9 +176,10 @@ protected boolean compareStructVectors(NonNullableStructVector left) {
       return false;
     }
 
+    RangeEqualsVisitor visitor = new RangeEqualsVisitor();
     for (String name : left.getChildFieldNames()) {
-      RangeEqualsVisitor visitor = new RangeEqualsVisitor(rightVector.getChild(name),
```

Comment:
also for compareListVectors/compareFixedSizeListVectors?

Reply:
Thanks for the comments. It seems the problem does not exist for compareListVectors/compareFixedSizeListVectors, because the visitor is created only once there.